Userland/Libraries/LibWeb/HTML/EventLoop/EventLoop.h

```diff
@@ -38,6 +38,7 @@ class EventLoop {
     TaskQueue const& microtask_queue() const { return m_microtask_queue; }
 
     void spin_until(JS::SafeFunction<bool()> goal_condition);
+    void spin_until_processing_tasks_with_source(Task::Source, JS::SafeFunction<bool()> goal_condition);
```

Comment:
the order of the operations in the function name doesn't match the argument order. `spin_processing_tasks_with_source_until` ? Putting the task source after a lambda in the caller would be awkward.